src/js/components/Form/__tests__/Form-test-uncontrolled.js

```diff
@@ -302,75 +392,212 @@ describe('Form uncontrolled', () => {
+          <FormField
+            onFocus={onFocus}
+            label="Name"
+            name="name"
+            placeholder="name"
+            required
+            validate={[
+              { regexp: /^[a-z]/i },
+              name => {
+                if (name && name.length === 1) return 'must be >1 character';
+                return undefined;
+              },
+              name => {
+                if (name === 'good')
+                  return {
+                    message: 'good',
+                    status: 'info',
+                  };
+                return undefined;
+              },
+            ]}
+          />
+
+          <FormField onFocus={onFocus} label="Email" name="email" required>
+            <TextInput
+              a11yTitle="test"
+              name="email"
+              type="email"
+              placeholder="email"
+            />
+          </FormField>
+          <Button onFocus={onFocus} label="submit" type="submit" />
+        </Form>
       </Grommet>,
     );
 
     // both fields have required error message
+    getByText('submit').focus();
     fireEvent.click(getByText('submit'));
     expect(queryAllByText('required')).toHaveLength(2);
 
     // one fields has required error message
+    getByPlaceholderText('name').focus();
     fireEvent.change(getByPlaceholderText('name'), {
       target: { value: 'Input has changed' },
     });
+    getByText('submit').focus();
     fireEvent.click(getByText('submit'));
     expect(queryAllByText('required')).toHaveLength(1);
 
-    // name field has new error and email field still
-    // has required error message
+    // name field has new error and email field still has required error message
+    getByPlaceholderText('name').focus();
     fireEvent.change(getByPlaceholderText('name'), {
       target: { value: 'a' },
     });
+    getByText('submit').focus();
     fireEvent.click(getByText('submit'));
     expect(queryByText('required')).toBeTruthy();
     expect(queryByText('must be >1 character')).toBeTruthy();
 
     //  new value in name does not remove the error message in email
+    getByPlaceholderText('name').focus();
     fireEvent.change(getByPlaceholderText('name'), {
       target: { value: 'abc' },
     });
+    expect(onFocus).toBeCalledTimes(6);
     expect(queryByText('required')).toBeTruthy();
     expect(queryByText('must be >1 character')).toBe(null);
   });
 
+  test('uncontrolled without name', () => {
+    const onSubmit = jest.fn();
+    const { getByPlaceholderText, getByText, container } = render(
+      <Form onSubmit={onSubmit}>
+        <FormField>
+          <TextInput a11yTitle="test" placeholder="test input" />
+        </FormField>
+        <Button type="submit" primary label="Submit" />
+      </Form>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'v' },
+    });
+    expect(container.firstChild).toMatchSnapshot();
```

Comment:
Can we add an assertion on the fetched input instead of taking the snapshot, unless you are positive the snapshot is capturing the new value.

Reply:
Removed snapshot and added assertion